src/__tests__/linter.test.ts

```diff
@@ -56,6 +56,43 @@ describe('linter', () => {
     return expect(spectral.run('123')).resolves.toBeTruthy();
   });
 
+  test('should run all rules despite of invalid JSON path expressions', () => {
+    spectral.setRules({
+      rule1: {
+        given: '$.bar[?(@.in==foo)]',
+        then: {
+          function: 'truthy',
+        },
+      },
+      rule2: {
+        given: '$.foo',
+        then: {
+          function: 'truthy',
+        },
+      },
+    });
+
+    return expect(
+      spectral.run(
+        {
+          bar: {
+            in: {},
+          },
+          foo: null,
+        },
+        { ignoreUnknownFormat: true },
+      ),
+    ).resolves.toEqual([
```

Comment:
This would reject entirely otherwise.

Reply:
@P0lip How about making the test function async (which makes the test slightly more readable)?

```
  test('should run all rules despite of invalid JSON path expressions', async () => {
    spectral.setRules({
      rule1: {
        given: '$.bar[?(@.in==foo)]',
        then: {
          function: 'truthy',
        },
      },
      rule2: {
        given: '$.foo',
        then: {
          function: 'truthy',
        },
      },
    });

    const res = await spectral.run(
      {
        bar: {
          in: {},
        },
        foo: null,
      },
      { ignoreUnknownFormat: true },
    );

    expect(res).toEqual([
      {
        code: 'rule2',
        message: '`foo` property is not truthy',
        path: ['foo'],
        range: expect.any(Object),
        severity: DiagnosticSeverity.Warning,
      },
    ]);

  });
```

Reply:
@nulltoken done